### PR TITLE
identity: bind identity service to did

### PR DESCRIPTION
**Pull request by manuelpolzhofer**
-  bind identity service to did

### CODE REVIEW

File: identity/did/bootstrapper.go [hunk 1]
```diff
@@ -48,7 +48,14 @@ func (*Bootstrapper) Bootstrap(context map[string]interface{}) error {
...
 
-	identity := NewIdentity(cfg, client)
+	//TODO config needs to store correct did format
+	id, err := cfg.GetIdentityID()
+	if err != nil {
+		return err
+	}
+	did := NewDID(common.BytesToAddress(id))
+
+	identity := NewIdentity(cfg, client, &did)
```
Comment (mikiquantum) [inline identity/did/bootstrapper.go]: There will be multiple identities under the same host. Why do we need to bootstrap here? Is this `identity` only used by the p2p layer? In which case we should rename it.

Comment (manuelpolzhofer) [reply identity/did/bootstrapper.go]: I thought multi-account is abstracted in that way, that the config already includes the identity address of the current account. 

I thought the bootstrapper would be called for every different identity in the multi-account scenario. If not I will remove it.

I added it to the bootstrapper to easier use the identity later on.

Comment (mikiquantum) [reply identity/did/bootstrapper.go]: > I thought multi-account is abstracted in that way, that the config already includes the identity address of the current account.

Not at bootstrap level. Accounts are dynamically created, loaded per request basis.

> I thought the bootstrapper would be called for every different identity in the multi-account scenario. If not I will remove it.

Nope, bootstrapper is only called at start up time. Then dynamically load the accounts per request. I would remove this as it is confusing. Within your code you should use the `context.Context` to identify the ID to use.

Comment (manuelpolzhofer) [reply identity/did/bootstrapper.go]: I already looked it up. Thanks for the clarification.

Comment (manuelpolzhofer) [reply identity/did/bootstrapper.go]: I will add the changes for context in the next PR

**Pull request merged**